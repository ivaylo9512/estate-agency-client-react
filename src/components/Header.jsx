import { useSelector } from "react-redux";
import Link from "next/link";
import { useEffect } from "react";

const { useDispatch } = require("react-redux");
const { removeUser } = require("../app/slicers/authenticate");

const Header = () => {
    const dispatch = useDispatch();
    const isAuth = useSelector(state => state.authenticate.isAuth);

    const logout = () => {
        dispatch(removeUser())
        localStorage.removeItem('user');
        localStorage.removeItem('Authorization');
    }

    return(
        <header>
            <Link href='/'>home</Link>
            <Link href='/properties'>properties</Link>
            {isAuth ?
                <>
                    <Link href='/create-property'>create</Link>
                    <button onClick={logout}>logout</button>
                </>
                :
                <>
                    <Link href='/login'>login</Link>
                    <Link href='/register'>register</Link>
                </>
            }
        </header>
    )
}
export default Header