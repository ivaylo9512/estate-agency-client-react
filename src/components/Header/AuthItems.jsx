import Link from 'next/link';

const AuthItems = () => {
    return(
        <>
            <li>
                <Link data-testid='login' href='/login'>login</Link>
            </li>
            <li>
                <Link data-testid='register' href='/register'>register</Link>
            </li>
        </>
    )
}

export default AuthItems