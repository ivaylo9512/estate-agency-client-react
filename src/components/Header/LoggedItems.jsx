import Link from 'next/link';

const LoggedItems = ({ logout }) => {
    return(
        <>
            <li>
                <Link data-testid='create-property' href='/create-property'>create</Link>
            </li>
            <li>
                <Link data-testid='account' href='/account'>account</Link>
            </li>
            <li>
                <button data-testid='logout' onClick={logout}>logout</button>
            </li>
        </>
    )
}

export default LoggedItems