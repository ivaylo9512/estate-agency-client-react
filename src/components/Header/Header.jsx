import { useSelector, useDispatch } from 'react-redux';
import Link from 'next/link';
import { getIsAuth, onLogout } from 'app/slices/authenticateSlice';
import LoggedItems from './LoggedItems';
import AuthItems from './AuthItems';

const Header = () => {
    const dispatch = useDispatch();
    const isAuth = useSelector(getIsAuth);

    const logout = () => {
        dispatch(onLogout())
    }

    return(
        <header>
            <nav>
                <ul>
                    <li>
                        <Link href='/'>home</Link>
                    </li>
                    <li>
                        <Link href='/properties'>properties</Link>
                    </li>
                    {isAuth 
                        ? <LoggedItems logout={logout}/>
                        : <AuthItems />
                    }
                </ul>
            </nav>
        </header>
    )
}
export default Header