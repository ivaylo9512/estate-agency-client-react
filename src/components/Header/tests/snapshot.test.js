import Header from '../Header';
import * as Redux from 'react-redux';
import { shallow } from 'enzyme';

describe('Header snapshot tests', () => {
    let selectorSpy;

    beforeAll(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());

        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (isAuth) => {
        selectorSpy.mockReturnValue(isAuth);

        return shallow(
            <Header />
        )
    }

    it('should match snapshot with isAuth to true', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with isAuth to false', () => {
        const wrapper = createWrapper(false);

        expect(wrapper).toMatchSnapshot();
    })
})