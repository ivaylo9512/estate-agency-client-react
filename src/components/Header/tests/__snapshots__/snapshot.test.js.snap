// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot tests should match snapshot with isAuth to false 1`] = `
<header>
  <nav>
    <ul>
      <li>
        <Link
          href="/"
        >
          home
        </Link>
      </li>
      <li>
        <Link
          href="/properties"
        >
          properties
        </Link>
      </li>
      <li>
        <Link
          data-testId="login"
          href="/login"
        >
          login
        </Link>
      </li>
      <li>
        <Link
          data-testId="register"
          href="/register"
        >
          register
        </Link>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header snapshot tests should match snapshot with isAuth to true 1`] = `
<header>
  <nav>
    <ul>
      <li>
        <Link
          href="/"
        >
          home
        </Link>
      </li>
      <li>
        <Link
          href="/properties"
        >
          properties
        </Link>
      </li>
      <li>
        <Link
          data-testId="create-property"
          href="/create-property"
        >
          create
        </Link>
      </li>
      <li>
        <Link
          data-testId="account"
          href="/account"
        >
          account
        </Link>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          logout
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
